diesel_infer_schema/src/information_schema.rs

```diff
@@ -77,6 +89,12 @@ pub fn determine_column_type(attr: &ColumnInformation) -> Result<ColumnType, Box
         &attr.type_name
     };
 
+    let tpe = if let Some(idx) = tpe.find('(') {
```

Comment:
This whole `determine_column_type` function got even weirder just now. Pulling it put might be a good idea. While you're at it, can you also add a test or two for its behavior (detecting arrays, things like`int(11)`, and the combination of the two if applicable)?